src/posts/controllers/posts.controller.ts

```diff
@@ -8,38 +8,78 @@ export class PostsController {
   constructor(private readonly postsService: PostsService) {}
 
   @Get()
-  findAll() {
-    return this.postsService.findAll();
+  async findAll() {
+    try {
+      return await this.postsService.findAll();
+    } catch (error) {
+        throw new NotFoundException(error); 
+    }
   }
 
   @Get('admin')
-  findAllAdmin() {
-    return this.postsService.findAll();
+  async findAllAdmin() {
+    try {
+      return await this.postsService.findAll();
+    } catch (error) {
+        throw new NotFoundException(error); 
+    }
   }
 
   @Get('search')
-  filter(@Query('term') term: string) {
-    return this.postsService.filter(term);
+  async filter(@Query('term') term: string) {
+    try {
+      return await this.postsService.filter(term);
+    } catch (error) {
+        throw new NotFoundException(error); 
+    }
   }
 
   @Get(':id')
-  findOne(@Param('id') id: string) {
-    return this.postsService.findOne(id);
+  async findOne(@Param('id') id: string) {
+    try {
+      return await this.postsService.findOne(id);
+    } catch (error) {
+        throw new NotFoundException(error);
+    }
   }
 
   @Post()
-  create(@Body() post: CreatePostDto) {
-    return this.postsService.create(post);
+  async create(@Body() post: CreatePostDto) {
+    try {
+      await this.postsService.create(post);
+      return 'Post criado com sucesso!'
+    } catch (error) {
+        throw new BadRequestException(error);
+    }
   }
 
   @Put(':id')
-  update(@Param('id') id: string, @Body() updatePostDto: UpdatePostDto) {
-    return this.postsService.update(id, updatePostDto);
+  async update(@Param('id') id: string, @Body() updatePostDto: UpdatePostDto) {
+    const isValidPost = await this.postsService.findOne(id);
+    if(isValidPost) {
+      try {
+        await this.postsService.update(id, updatePostDto);
+        return 'Post atualizado com sucesso!'
+      } catch (error) {
+          throw new BadRequestException(error);
+      }
+    } else {
+      throw new NotFoundException();
+    }
   }
 
   @Delete(':id')
-  remove(@Param('id') id: string) {
-    return this.postsService.remove(id);
+  async remove(@Param('id') id: string) {
+    const isValidPost = await this.postsService.findOne(id);
+    if(isValidPost) {
+      try {
+        await this.postsService.remove(id);
+        return 'Post excluído com sucesso!'
+      } catch (error) {
+          throw new BadRequestException(error);
+      }
+    } else {
+      throw new NotFoundException();
+    }
```

Comment:
Já que estamos usando um `try` `catch` aqui, pensando se não seria melhor colocar tudo dentro dele já. E aí o erro do `catch` não poderia ser `NotFoundException` porque  já validamos que o post existe com o `isValidPost`, então eu mudaria o erro pra `InternalServerErrorException`. E aí eu faria a mesma coisa pro `update`

```suggestion
    try {
        const isValidPost = await this.postsService.findOne(id);
        if(isValidPost) {
            await this.postsService.remove(id);
        } else {
            throw new NotFoundException();
        }
        return 'Post excluído com sucesso!'
      } catch (error) {
          throw new InternalServerErrorException(error);
      }

```